app/jobs/appointment_reminders_job.rb

```diff
@@ -0,0 +1,28 @@
+class AppointmentRemindersJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    Appointment.within_48_hours.each do |appointment|
+      send_reminder(appointment)
+    end
+  end
+
+  private
+
+  def send_reminder(appointment)
+    expected_reminder_count = appointment.audits.inject(0) do |count, audit|
```

Comment:
Exactly. I thought about the various issues and it seems to cover all the cases correctly as far as I could tell.